src/Zoomba/APILibrary.py

```diff
@@ -328,3 +315,37 @@ def generate_unmatched_keys_error_message(self, unmatched_keys):
                     keys_error_msg += str(key_error) + "\n"
             assert False, keys_error_msg + "\nPlease see differing value(s)"
 
+
+def _date_format(date_string, key, unmatched_keys_list, date_type, date_format=None):
```

Comment:
Why do we need this whole block for python 3?

Reply:
To my knowledge this is not a python3 specific change - it's a bug fix to the date format error handling. 

I would prefer if these were different PRs though. They are entirely separate issues.